superduperdb/base/datalayer.py

```diff
@@ -1076,6 +1087,16 @@ def infer_schema(
         """
         return self.databackend.infer_schema(data, identifier)
 
+    def set_cfg(self, cfg: Config):
```

Comment:
Why not do this?

```python
@cfg.setter
def cfg(self, value):
   ...
```

Reply:
Also, do we really want to do this? That would mess things up: `self.compute` depends on the config used to build the `Datalayer`. So you set the `cfg` and it doesn't match the attributes of the `db`.

Reply:
There should be an original issue here. If we start a datalayer by invoking superduperdb(**new_config), the value of the config is actually incorrect at this point because it has not been updated using **new_config. Therefore, when we send tasks to the compute layer, we should send the config with which we actually constructed the datalayer, not the one from the configuration file; otherwise, there will be a mismatch between the client side and the compute side.

I believe that the correct approach should be to launch the datalayer using the received CFG in each compute job, rather than merely retrieving it from a configuration file.

WDYT?